### PR TITLE
Models connecting as per Issue #5

### PR DESCRIPTION
**Pull request by SupriyaGadigone**
Working on #18. Is there more to it than this?
I wanted to write some tests cases on this too but I guess I need the repositories to be merged so I can do it after.

### CODE REVIEW

File: src/main/java/SYSC4806/Model/Course.java
Comment (K-shamG) [inline src/main/java/SYSC4806/Model/Course.java]: Referencing https://www.baeldung.com/spring-data-rest-relationships, I think this needs to have a CascadeType.ALL and it looks like they use @JoinTable to create that intermediary table for many to many relationships.

File: src/main/java/SYSC4806/Model/Program.java
Comment (K-shamG) [inline src/main/java/SYSC4806/Model/Program.java]: I believe this should be a @ManyToMany as a course can belong to many programs (also in Course, you specified this relationship as such). Following the same tutorial, I think this needs a @mappedBy attribute.

File: src/main/java/SYSC4806/Model/LearningOutcome.java
Comment (K-shamG) [inline src/main/java/SYSC4806/Model/LearningOutcome.java]: In category, you have it listed as one-to-many. A learning outcome only maps to one category; this should be a many-to-one.

**Review by K-shamG — CHANGES_REQUESTED** [review]
As you said, we need to wait for Ryan's pull request to get merged before you can test it. We can have someone work on testing this branch with you, but I'd prefer to have tests written before it's merged since this is a crucial part to having it work.

**Comment by RyanWork** [comment]
#25 is merged now. You can try implementing it now as Kshamina said

**Comment by SupriyaGadigone** [comment]
Bunch of things changed here:
1. The data type for the entities were ArrayList before so I had to change it to List since ArrayList is not supported (https://stackoverflow.com/questions/44157641/which-collections-are-supported-by-hibernate-manytomany-annotation)
2. In the LearningOutcome class the category had to be changed from a List to just Category since its @ManyToOne mapping (followed kshamina's link https://www.baeldung.com/spring-data-rest-relationships)  
3. Added the META-INF folder with persistence.xml file

Sorry guys, I dont know how I missed all of this basic stuff, I will follow the labs properly from next time 

Anyways, next up will be tests for all of this.

**Review by K-shamG — APPROVED** [review]
I'm good with this for now since I see tests are in another issue. It can always be modified after testing.

**Review by RyanWork — APPROVED** [review]

**Comment by SupriyaGadigone** [comment]
Bunch of things here:
- Fixed the mappings as per the tests 
- The tests are very basic where they test all the methods in the Repository classes created by Ryan. I am basically just creating dummy objects, saving them and then testing if they persist to see if the mappings work.

We can add more tests after as well.

**Comment by SupriyaGadigone** [comment]
Also, if you don't want to test it yourself, proof that the test passed :P 
screen shot 2019-02-28 at 12 49 34 pm

**Comment by RyanWork** [comment]
I think we should consider initializing some properties in our empty constructor. Whether or not this is null (or a default of some primitive), it doesn't hurt to do that? For example, we might want to initialize the Lists<> to a default empty list and not necessarily leave it null. We could call the default constructors like:
```java
public Course(){
    this.Course(0, null, null, new ArrayList<LearningOutcome>(), new ArrayList<Program>());
}
```

Edit; this might be relevant to #35 so this could be an issue for later after we define something.

**Review by RyanWork — APPROVED** [review]

**Review by andrewdlqnguyen — APPROVED** [review]

**Pull request merged**
